lib/fs.js

```diff
@@ -1112,14 +1114,14 @@ function writeAll(fd, buffer, offset, length, position, callback) {
 }
 
 fs.writeFile = function(path, data, options, callback) {
-  var callback = maybeCallback(arguments[arguments.length - 1]);
+  callback = maybeCallback(arguments[arguments.length - 1]);
```

Comment:
it is easy to prevent just by renaming the parameter to something like `callback_` and leaving the var keyword where it was.